queue_job/models/base.py

```diff
@@ -27,6 +27,20 @@ def _register_hook(self):
         for job_method in job_methods:
             self.env['queue.job.function']._register_job(self, job_method)
 
+    @api.model
+    def get_job_context_keys(self):
```

Comment:
these methods should be private IMO

Reply:
Suggestion: setup the context keys to pass on the @job decorator

```
@job(context_allow=['foo'])
```

With `['force_company', 'lang', 'tz']` by default and new keys added to the list.
The filtering of the context keys could then be hidden in the job implementation details, instead of cluttering the `base` model (I try hard to put as less method as possible here).

Would it work for you?

Reply:
Or if the safe list really needs to be global, then it can be added to `queue.job`

Reply:
great idea about the job. I will make the change